daemon/container_operations_unix.go

```diff
@@ -772,6 +768,9 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 	}
 
 	if endpointConfig != nil {
+		if !containertypes.NetworkMode(idOrName).IsUserDefined() && len(endpointConfig.Aliases) > 0 {
+			return runconfig.ErrUnsupportedNetworkAndAlias
+		}
```

Comment:
I think it's better to also move `if err := validateNetworkingConfig(n, endpointConfig); err != nil {` here